client-reactjs/src/components/application/actions/LandingZoneUpload.jsx

```diff
@@ -0,0 +1,337 @@
+  const handleFileUpload = () =>{
+    let newFiles = files.map(file => file.name);
+    let {machine, pathToAsset} = fileDestinationDetail;
+    let commonFiles = fileDestinationDetail.currentDirectoryFiles.filter(file => newFiles.includes(file));
+    if(!cluster){
+      message.error("Select a cluster")
+    }else if(!pathToAsset){
+      message.error("Select  destination folder")
+    }
+    else if(files.length < 1){
+      message.error("Please select atleast one file to upload")
+    }else if(commonFiles.length > 0 && !fileDestinationDetail.overWriteFiles){
+      message.error("Some file(s) already exist. Please check overwrite box to continue")
+    }
+    else{
+      setUploading(true);
+      setFiles(prev => (prev.map(item => { 
+      item.uploadStatus = 'uploading'; return item})));
+
+    // Start by sending some file and destination details to server
+    socket.emit('start-upload', {pathToAsset, cluster, machine});
+    files.forEach(file => {
+      const {uid : id, name : fileName, size : fileSize} = file.originFileObj;
+      if(file.size <= 1000000){
+         let reader = new FileReader();
+         reader.readAsArrayBuffer(file.originFileObj);
+          reader.onload = function(e){
+            let arrayBuffer = e.target.result;
+            socket.emit('upload-file', {
+              id, fileName,
+              data: arrayBuffer
+            })
+          }
+      }else{
+        let slice = file.originFileObj.slice(0, 100000);
+        let reader = new FileReader();
+        reader.readAsArrayBuffer(slice);
+        reader.onload = function(e){
+          let arrayBuffer = e.target.result;
+          socket.emit('upload-slice', {
+            id, fileName,
+            data: arrayBuffer,
+            fileSize,
+            chunkStart : 100000,
+            chunkSize: 100000
+          })
+        }
+      }
+    })
+
+     //When server asks for a slice of a file
+     socket.on('supply-slice', (message) =>{
+      let currentFile = files.find(file => file.originFileObj.uid === message.id);
+      let slice = currentFile.originFileObj.slice(message.chunkStart, message.chunkStart + message.chunkSize);
+      let reader = new FileReader();
+      reader.readAsArrayBuffer(slice);
+      reader.onload = function(e){
+        let arrayBuffer = e.target.result;
+        socket.emit('upload-slice', {
+          id: currentFile.uid,
+          fileName : currentFile.name,
+          data: arrayBuffer,
+          fileSize : currentFile.size,
+          chunkSize : message.chunkSize
+        })
+      }
+    })
+    }
+  }
+
+ //Dragger props
+  const props = {
+    name: 'file',
+    multiple: true,
+    showUploadList: false,
+    accept: '.xls, .xlsm, .xlsx, .txt, .json, .csv',
+    onChange(info) {
+      const { status } = info.file;
+      setFiles(() =>info.fileList.slice(-5).map(file => {file.key = file.uid; return file;})); //Causing issue -> file list should be used
+     if (status === 'error') {
```

Comment:
this is a bit consufing, if status is error will there still be a info.fileList? I think we should check status before and exit the function without updating state at all, if status is done only then we need to update setFiles. Also there is no need to create key value on each file object, it can be done on table :
``` <Table
      //... rest fields
      rowKey={(record) => record.uid} /> ```